Crash, heap dump  missing

HeapAnalyzer crashes because the heap dump file passed to the service didn't exist

```
leakcanary.HeapAnalysisException: java.lang.IllegalArgumentException: File does not exist: /storage/emulated/0/Download/leakcanary-com.squareup.development/f871e570-be04-4368-9621-c964c27671e8_pending.hprof
        at leakcanary.HeapAnalyzer.checkForLeaks(HeapAnalyzer.kt:93)
        at leakcanary.internal.HeapAnalyzerService.onHandleIntentInForeground(HeapAnalyzerService.kt:49)
        at leakcanary.internal.ForegroundService.onHandleIntent(ForegroundService.kt:55)
        at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:78)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:214)
        at android.os.HandlerThread.run(HandlerThread.java:67)
Caused by: java.lang.IllegalArgumentException: File does not exist: /storage/emulated/0/Download/leakcanary-com.squareup.development/f871e570-be04-4368-9621-c964c27671e8_pending.hprof
        at leakcanary.HeapAnalyzer.checkForLeaks(HeapAnalyzer.kt:90)
        at leakcanary.internal.HeapAnalyzerService.onHandleIntentInForeground(HeapAnalyzerService.kt:49)
        at leakcanary.internal.ForegroundService.onHandleIntent(ForegroundService.kt:55)
        at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:78)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:214)
        at android.os.HandlerThread.run(HandlerThread.java:67)
```

Not sure how this can happen. Worth noticing that 2 seconds before we were successfully parsing and handling a hprof (same? another one?)

The latest release of leakcanary does not block if an analysis is in progress, so there might be concurrency issue or old cleanup code that's too aggressive.

--- a/leakcanary-android-core/src/main/java/leakcanary/DefaultAnalysisResultListener.kt
+++ b/leakcanary-android-core/src/main/java/leakcanary/DefaultAnalysisResultListener.kt
@@ -86,12 +86,14 @@ object DefaultAnalysisResultListener : AnalysisResultListener {
     val fileName = SimpleDateFormat("yyyy-MM-dd_HH-mm-ss_SSS'.hprof'", Locale.US).format(Date())
 
     val newFile = File(heapDumpFile.parent, fileName)
+    val lastModified = heapDumpFile.lastModified()
     val renamed = heapDumpFile.renameTo(newFile)
     if (!renamed) {
       CanaryLog.d(
           "Could not rename heap dump file %s to %s", heapDumpFile.path, newFile.path
       )
     }
+    newFile.setLastModified(lastModified)
     return newFile
   }
 }

